sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/main/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobCheckpointStore.java

```diff
@@ -66,22 +68,70 @@ public BlobCheckpointStore(BlobContainerAsyncClient blobContainerAsyncClient) {
      * from the Storage Blobs. Could return empty results if there are is no existing ownership information.
      *
      * @param eventHubName The Event Hub name to get ownership information.
-     * @param consumerGroupName The consumer group name.
+     * @param consumerGroup The consumer group name.
      * @return A flux of partition ownership details of all the partitions that have/had an owner.
      */
     @Override
     public Flux<PartitionOwnership> listOwnership(String fullyQualifiedNamespace, String eventHubName,
-        String consumerGroupName) {
-        String prefix = getBlobPrefix(eventHubName, consumerGroupName);
+        String consumerGroup) {
+        String prefix = getBlobPrefix(fullyQualifiedNamespace, eventHubName, consumerGroup, OWNERSHIP_PATH);
         BlobListDetails details = new BlobListDetails().setRetrieveMetadata(true);
         ListBlobsOptions options = new ListBlobsOptions().setPrefix(prefix).setDetails(details);
         return blobContainerAsyncClient.listBlobs(options)
-            // Blob names should be of the pattern eventhub/consumergroup/<partitionId>
+            // Blob names should be of the pattern
+            // fullyqualifiednamespace/eventhub/consumergroup/ownership/<partitionId>
             // While we can further check if the partition id is numeric, it may not necessarily be the case in future.
-            .filter(blobItem -> blobItem.getName().split(BLOB_PATH_SEPARATOR).length == 3)
+            .filter(blobItem -> blobItem.getName().split(BLOB_PATH_SEPARATOR).length == 5)
```

Comment:
Could this be changed to a `flatMap` or `map` where we split the name into a `String[]` then if there are 5 pieces we initiate the conversion to partition ownership? This current implementation will split the name twice and that could be optimized slightly.